src/main/java/com/example/demo/service/CartService.java

```diff
@@ -27,16 +35,16 @@ public Cart createCart() {
     public Cart addProductToCart(Long cartId, Long productId) {
         log.info("Adding product with ID {} to cart with ID {}", productId, cartId);
         Cart cart = cartRepository.findById(cartId)
-                .orElseThrow(() -> new RuntimeException("Cart not found"));
+                .orElseThrow(() -> new CartNotFoundException("Cart not found"));
 
+        validateCartStatus(cart);
         Optional<Long> productInfo = productServiceFeignClient.getProductInfo(productId);
 
         if (productInfo.isEmpty()) {
-            throw new RuntimeException("Product not found");
+            throw new ProductNotFoundException("Product not found");
         }
 
         Long productID = productInfo.get();
-
         if (cart.getProductIds().contains(productID)) {
             throw new RuntimeException("Product already added to the cart");
```

Comment:
a co w przypadku, gdy np. bedziemy chcieli dodac dwa takie same produkty do koszyka?

Reply:
+1

Reply:
+2

Reply:
+3

Reply:
To wtedy jeżeli będzie to ten sam produkt, to nikt ci nie zabroni dodać tego samego produktu, nie możesz dodać tego samego id produktu.